File: pset1/mario/mario.c
#include <stdio.h>
#include <cs50.h>

void draw_pyramid(int height);

int main(void){
    int height;
    do
    {
        height = get_int("Height:");
    }while (height < 1 || height >8);
    draw_pyramid(height);
}

void draw_pyramid(int height)
{   
    int hashes = 0;

    for(int i = 0; i < height; i++)
    {   
        hashes++;
        int white_space = height - hashes;

        // print white space
        for(int j = 0; j < white_space; j++)
        {
            printf(" ");
        }
        
        // print hashes
        for(int x = 0; x < hashes; x++)
        {
            printf("#");
        }

        // print gap
        printf("  ");

        //print hashes
        for(int y = 0; y < hashes; y++)
        {
            printf("#");
        }       
        printf("\n");
        
    }
}

File: pset6/mario/more/mario.py
from cs50 import get_int

while True:
    pyramid_height = get_int("Pyramid height: ")
    hashes = 0
    if pyramid_height > 8 or pyramid_height < 1:
        continue
    else:
        for i in range(pyramid_height):
            hashes += 1
            white_space = pyramid_height - hashes
            
            # print first half
            print(" " * white_space, end="")
            print("#" * hashes, end="")
            
            # print space between
            print(" " * 2, end="")
            
            # print second half
            print("#" * hashes, end="")
            print()
        break
